src/Platforms/OpenAI/ProductMapper.php

```diff
@@ -838,43 +771,54 @@ protected function get_pickup_sla(): ?string {
 	}
 
 	/**
-	 * Get product warning.
+	 * Get related product ID.
 	 *
-	 * @param \WC_Product $product Product object.
-	 * @return string|null Product warning or null.
-	 */
-	protected function get_warning( \WC_Product $product ): ?string {
-		return $this->get_meta_value( $product, '_wpfoai_warning' );
-	}
-
-	/**
-	 * Get product warning.
+	 * Returns IDs from upsell or cross-sell products as a comma-separated list.
+	 * Prioritizes upsell products over cross-sell products.
 	 *
 	 * @param \WC_Product $product Product object.
-	 * @return string|null Product warning or null.
+	 * @return string|null Comma-separated list of related product IDs or null.
 	 */
-	protected function get_warning_url( \WC_Product $product ): ?string {
-		return $this->get_meta_value( $product, '_wpfoai_warning_url' );
-	}
+	protected function get_related_product_id( \WC_Product $product ): ?string {
+		$upsell_ids     = $product->get_upsell_ids();
+		$cross_sell_ids = $product->get_cross_sell_ids();
 
-	/**
-	 * Get product age restriction.
-	 *
-	 * @param \WC_Product $product Product object.
-	 * @return string|null Product age restriction or null.
-	 */
-	protected function get_age_restriction( \WC_Product $product ): ?string {
-		return $this->get_meta_value( $product, '_wpfoai_age_restriction' );
+		// Prioritize upsell over cross-sell.
+		if ( ! empty( $upsell_ids ) ) {
```

Comment:
Alternatively, instead of prioritize one over the other, we could merge arrays or return a concatenated string.